cluster/storage/src/main/java/com/linkedin/openhouse/cluster/storage/local/LocalStorage.java

```diff
@@ -0,0 +1,70 @@
+package com.linkedin.openhouse.cluster.storage.local;
+
+import com.linkedin.openhouse.cluster.storage.Storage;
+import com.linkedin.openhouse.cluster.storage.StorageClient;
+import com.linkedin.openhouse.cluster.storage.StorageType;
+import com.linkedin.openhouse.cluster.storage.configs.StorageProperties;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Component;
+
+/**
+ * The LocalStorage class is an implementation of the Storage interface for local storage. It uses a
+ * LocalStorageClient to interact with the local file system.
+ */
+@Component
+public class LocalStorage implements Storage {
+
+  private static final StorageType.Type TYPE = StorageType.LOCAL;
+
+  @Autowired private StorageProperties storageProperties;
+
+  // Lazy initialization of the LocalStorageClient
+  @Autowired @Lazy private LocalStorageClient localStorageClient;
+
+  /**
+   * Check if the local storage is active.
+   *
+   * <p>The local storage is considered active if the default type is not set or no types are
+   * provided or specific "local" type is provided.
+   *
+   * @return true if the local storage is active, false otherwise
+   */
+  @Override
+  public boolean isActive() {
+    if (storageProperties.getDefaultType() == null) {
+      return true;
+    } else if (storageProperties.getTypes() == null || storageProperties.getTypes().isEmpty()) {
+      return true;
+    } else {
+      return storageProperties.getTypes().containsKey(TYPE.getValue());
+    }
+  }
+
+  /**
+   * Get the properties of the local storage.
+   *
+   * @return a copy map of properties of the local storage
+   */
+  @Override
```

Comment:
This code will likely be common to all storage types? Should it be in a base class then? It is ok to add it here for now and when we add the second storage type, we can move code to avoid duplication.

Reply:
Yes, we could pursue it in the next PR!